internal/discovery/secret_lister.go

```diff
@@ -6,27 +6,46 @@ import (
 	listersv1 "k8s.io/client-go/listers/core/v1"
 )
 
+type SecretListerer interface {
+	Lister(namespace string) (listersv1.SecretNamespaceLister, bool)
+}
+
 type SecretListerConfig struct {
 	// Namespace can be used to restric the search to a particular namespace.
-	Namespace string
+	Namespaces []string
 	// Client is the Kubernetes client.Interface used to build informers.
 	Client kubernetes.Interface
 }
 
+type MultiNamespaceSecretListerer struct {
+	listers map[string]listersv1.SecretNamespaceLister
+}
+
+func (l MultiNamespaceSecretListerer) Lister(namespace string) (listersv1.SecretNamespaceLister, bool) {
+	lister, ok := l.listers[namespace]
+
+	return lister, ok
+}
+
 // NewSecretNamespaceLister returns a SecretGetter to get secrets with informers.
-func NewSecretNamespaceLister(config SecretListerConfig) (listersv1.SecretNamespaceLister, chan<- struct{}) {
+func NewSecretNamespaceLister(config SecretListerConfig) (*MultiNamespaceSecretListerer, chan<- struct{}) {
 	stopCh := make(chan struct{})
 
 	factory := informers.NewSharedInformerFactoryWithOptions(
 		config.Client,
 		defaultResyncDuration,
-		informers.WithNamespace(config.Namespace),
```

Comment:
I fear that with this change the permission needed is to watch the secrets over the all clusters, before we needed such permission over the single namespace selected 😕  

The helm chart however is not smart enough to make any difference

Reply:
creating a `informers.NewSharedInformerFactoryWithOptions` per namespace would be a solution... would it worth the overhead?

Reply:
my understanding is that the only namespace that is going to be get,listed,watched is the one we create a lister for (few lines below).
I still need to add some test to this , I can test that too.

Reply:
yes your fear was confirmed 😢 is not working as I expected. 
I need to build an informer/namespace i think

Reply:
added some test and include the multi factory thing , it doesn't look to complex in the code and we could use the same closer at least